src/utils/enhanceDevice.ts

```diff
@@ -0,0 +1,35 @@
+import { omit, concat, assign, reduce, isEmpty } from "lodash-es";
+import type { DeviceType } from "../types";
+import type { Site, Device, DevicesResponse, SearchResponce } from "../services/lansweeper/types";
+
+const enhance = (devices: Device[], site: Site) => {
+  return devices.map((d) => assign({}, d, { site }));
+};
+
+const enhanceSearchDevices = (site?: Site, data?: SearchResponce): DeviceType[] => {
+  const items = data?.site?.assetResources?.items;
+  if (isEmpty(site) || isEmpty(data) || !Array.isArray(items)) {
+    return [];
+  }
+
+  return items.map((d: Device) => {
+    return assign({}, d, { site }) as DeviceType;
+  });
+};
+
+const enhanceDevices = (data?: DevicesResponse): DeviceType[] => {
```

Comment:
This is probably meant to read `(data: DevicesResponse|undefined)` as `data` must always be sent in but it can be undefined. 

Personally I'd removed that complexity and instead just say that `data` must always be of type `DevicesResponse` and enforce the calling code sending in the right data.

Reply:
Understood your point. Changed to `(data: Maybe<SearchResponce>)`. Because raw data from the API response comes here.